ogr/ogrsf_frmts/adbc/ogradbcbigquerylayer.cpp

```diff
@@ -0,0 +1,981 @@
+    osSQL += "` ";
+    if (osFieldNames.empty())
+    {
+        osSQL += "DEFAULT VALUES";
+    }
+    else
+    {
+        osSQL += '(';
+        osSQL += osFieldNames;
+        osSQL += ") VALUES (";
+        osSQL += osFieldValues;
+        osSQL += ')';
+    }
+    return m_poDS->CreateInternalLayer(osSQL.c_str())->GotError()
+               ? OGRERR_FAILURE
+               : OGRERR_NONE;
+}
+
+/************************************************************************/
+/*                           ISetFeature()                              */
+/************************************************************************/
+
+OGRErr OGRADBCBigQueryLayer::ISetFeature(OGRFeature *poFeature)
+{
+    if (m_poDS->GetAccess() != GA_Update)
+    {
+        CPLError(
+            CE_Failure, CPLE_NotSupported,
+            "SetFeature() only supported on datasets opened in update mode");
+        return OGRERR_FAILURE;
+    }
+    if (m_osFIDColName.empty())
+    {
+        CPLError(CE_Failure, CPLE_NotSupported,
+                 "SetFeature() only supported on tables with a INT64 single "
+                 "column primary key");
+        return OGRERR_FAILURE;
+    }
+    if (poFeature->GetFID() < 0)
+    {
+        return OGRERR_NON_EXISTING_FEATURE;
+    }
+
+    if (!m_poAdapterLayer)
+        BuildLayerDefn();
+    if (m_bDeferredCreation)
+        RunDeferredCreation();
+    if (m_bLayerDefinitionError)
+        return OGRERR_FAILURE;
+
+    std::string osDatasetId;
+    std::string osTableId;
+    if (!STARTS_WITH_CI(m_osBaseStatement.c_str(), "SELECT * FROM ") ||
+        CPLString(m_osBaseStatement).ifind(" WHERE ") != std::string::npos ||
+        !GetBigQueryDatasetAndTableId(osDatasetId, osTableId))
+    {
+        CPLError(CE_Failure, CPLE_NotSupported,
+                 "SetFeature(): cannot get dataset and table ID");
+        return OGRERR_FAILURE;
+    }
+
+    std::string osSQL = "UPDATE `";
+    osSQL += OGRDuplicateCharacter(osDatasetId, '`');
+    osSQL += "`.`";
+    osSQL += OGRDuplicateCharacter(osTableId, '`');
+    osSQL += "` SET ";
+
+    bool bAddComma = false;
+    auto poFeatureDefn = GetLayerDefn();
+    for (int i = 0; i < poFeatureDefn->GetGeomFieldCount(); ++i)
+    {
+        const char *pszName = poFeatureDefn->GetGeomFieldDefn(i)->GetNameRef();
+        if (bAddComma)
+            osSQL += ", ";
+        bAddComma = true;
+        osSQL += '`';
+        osSQL += OGRDuplicateCharacter(pszName, '`');
+        osSQL += "` = ";
+        const auto poGeom = poFeature->GetGeomFieldRef(i);
+        if (poGeom)
+        {
+            osSQL += "ST_GeogFromText('";
+            char *pszWKT = nullptr;
+            poGeom->exportToWkt(&pszWKT, wkbVariantIso);
+            osSQL += pszWKT;
+            CPLFree(pszWKT);
+            osSQL += "')";
+        }
+        else
+        {
+            osSQL += "NULL";
+        }
+    }
+    for (int i = 0; i < poFeatureDefn->GetFieldCount(); ++i)
```

Comment:
```suggestion

    for (int i = 0; i < poFeatureDefn->GetFieldCount(); ++i)
```

TODO for self: check if we can add a rule in clang format.

Reply:
fixed